components/function-controller/internal/webhook/conversion_webhook_test.go

```diff
@@ -0,0 +1,365 @@
+								},
+								Requests: corev1.ResourceList{
+									corev1.ResourceCPU:    resource.MustParse("50m"),
+									corev1.ResourceMemory: resource.MustParse("64Mi"),
+								},
+							},
+						},
+					},
+					Runtime: serverlessv1alpha2.NodeJs12,
+					Source: serverlessv1alpha2.Source{
+						Inline: &serverlessv1alpha2.InlineSource{
+							Source:       "test-source",
+							Dependencies: "test-deps",
+						},
+					},
+				},
+			},
+			wantDst: &serverlessv1alpha1.Function{
+				ObjectMeta: metav1.ObjectMeta{Name: "test", Namespace: "test"},
+				Spec: serverlessv1alpha1.FunctionSpec{
+					Runtime: serverlessv1alpha1.Nodejs12,
+					Resources: corev1.ResourceRequirements{
+						Limits: corev1.ResourceList{
+							corev1.ResourceCPU:    resource.MustParse("100m"),
+							corev1.ResourceMemory: resource.MustParse("128Mi"),
+						},
+						Requests: corev1.ResourceList{
+							corev1.ResourceCPU:    resource.MustParse("50m"),
+							corev1.ResourceMemory: resource.MustParse("64Mi"),
+						},
+					},
+					BuildResources: corev1.ResourceRequirements{
+						Limits: corev1.ResourceList{
+							corev1.ResourceCPU:    resource.MustParse("300m"),
+							corev1.ResourceMemory: resource.MustParse("300Mi"),
+						},
+						Requests: corev1.ResourceList{
+							corev1.ResourceCPU:    resource.MustParse("200m"),
+							corev1.ResourceMemory: resource.MustParse("200Mi"),
+						},
+					},
+
+					Source: "test-source",
+					Deps:   "test-deps",
+				},
+			},
+			wantVersion: serverlessv1alpha1.GroupVersion.String(),
+		},
+		{
+			name: "v1alpha1 to v1alpha2 Git function",
+			src: &serverlessv1alpha1.Function{
+				ObjectMeta: metav1.ObjectMeta{Name: "test", Namespace: "test"},
+				Spec: serverlessv1alpha1.FunctionSpec{
+					Runtime: serverlessv1alpha1.Nodejs12,
+					Type:    serverlessv1alpha1.SourceTypeGit,
+					Source:  testRepoName,
+					Repository: serverlessv1alpha1.Repository{
+						BaseDir:   "/code/",
+						Reference: "main",
+					},
+				},
+			},
+			wantDst: &serverlessv1alpha2.Function{
+				ObjectMeta: metav1.ObjectMeta{
+					Name: "test", Namespace: "test",
+					Annotations: map[string]string{
+						v1alpha1GitRepoNameAnnotation: testRepoName,
+					},
+				},
+				Spec: serverlessv1alpha2.FunctionSpec{
+					Runtime: serverlessv1alpha2.NodeJs12,
+					Source: serverlessv1alpha2.Source{
+						GitRepository: &serverlessv1alpha2.GitRepositorySource{
+							URL: testGitRepo.Spec.URL,
+							Repository: serverlessv1alpha2.Repository{
+								BaseDir:   "/code/",
+								Reference: "main",
+							},
+							Auth: &serverlessv1alpha2.RepositoryAuth{
+								Type:       serverlessv1alpha2.RepositoryAuthType(testGitRepo.Spec.Auth.Type),
+								SecretName: testGitRepo.Spec.Auth.SecretName,
+							},
+						},
+					},
+				},
+			},
+			wantVersion: serverlessv1alpha2.GroupVersion.String(),
+		},
+		{
+			name: "v1alpha2 to v1alpha1 Git function",
```

Comment:
I would add the case with status conversion from v1 to v2.

Reply:
Good Call :+1: